src/c++/perf_analyzer/load_worker.h

```diff
@@ -254,6 +266,42 @@ class LoadWorker : public IWorker {
       const int step_index);
 
  protected:
+  /// A helper function to issue inference request to the server.
+  /// \param context InferContext to use for sending the request.
+  /// \param context_id The ID of the context
+  /// \param request_id The unique id to be associated with the request.
+  /// \param delayed Whether the request fell behind its scheduled time.
+  /// \param callback_func The callback function to use with asynchronous
+  /// request.
+  /// \param async_req_map The map from ongoing request_id to the
+  /// request information needed to correctly interpret the details.
+  /// \param thread_stat The runnning status of the worker thread
+  void Request(
+      std::shared_ptr<InferContext> context, const uint32_t ctx_id,
+      const uint64_t request_id, const bool delayed,
+      cb::OnCompleteFn callback_func,
+      std::map<std::string, AsyncRequestProperties>& async_req_map,
+      std::shared_ptr<ThreadStat> thread_stat);
+
+  // Callback function for handling asynchronous requests
+  void AsyncCallbackFuncImpl(cb::InferResult* result);
+
+  // Code to execute at the end of the async callback function
+  virtual void AsyncCallbackFinalize(uint32_t ctx_id) {}
```

Comment:
Should this be pure virtual or throw an exception?

Reply:
the rest of the code base throws cb::Errors::Failure errors. Maybe that?

Reply:
ConcurrencyWorker has real code, and RequestRateWorker has no code. The way I did that was an empty base an override in ConcurrencyWorker. If you think pure virtual with an empty implementation in RequestRateWorker is cleaner I can do that.

Reply:
I think pure virtual would be cleaner.

Reply:
Fixed with pure virtual